src/futils.cpp

```diff
@@ -502,4 +493,71 @@ namespace Exiv2 {
         const size_t idxLastSeparator = ret.find_last_of(EXV_SEPARATOR_STR);
         return ret.substr(0, idxLastSeparator);
     }
+
+    static bool pushPath(std::string& path,std::vector<std::string>& libs,std::set<std::string> & paths)
+    {
+        bool result = Exiv2::fileExists(path,true) && paths.find(path) == paths.end() && path != "/" ;
+        if ( result ) {
+            paths.insert(path);
+            libs.push_back(path);
+        }
+        return result ;
+    }
+
+    std::vector<std::string> getLoadedLibraries()
+    {
+        std::vector<std::string>  libs ;
+        std::set<std::string>     paths;
+        std::string               path ;
+
+    #if defined(WIN32) || defined(__CYGWIN__) || defined(__MINGW__)
+        // enumerate loaded libraries and determine path to executable
+        HMODULE handles[200];
+        DWORD   cbNeeded;
+        if ( EnumProcessModules(GetCurrentProcess(),handles,lengthof(handles),&cbNeeded)) {
+            char szFilename[_MAX_PATH];
+            for ( DWORD h = 0 ; h < cbNeeded/sizeof(handles[0]) ; h++ ) {
+                GetModuleFileNameA(handles[h],szFilename,lengthof(szFilename)) ;
+                std::string path(szFilename);
+                pushPath(path,libs,paths);
+            }
+        }
+    #elif defined(__APPLE__)
+        // man 3 dyld
+        uint32_t count = _dyld_image_count();
+        for (uint32_t image = 0 ; image < count ; image++ ) {
+            std::string path(_dyld_get_image_name(image));
+            pushPath(path,libs,paths);
+        }
+    #elif defined(__FreeBSD__)
+        unsigned int n;
+        struct procstat*      procstat = procstat_open_sysctl();
+        struct kinfo_proc*    procs    = procstat ? procstat_getprocs(procstat, KERN_PROC_PID, getpid(), &n) : NULL;
+        struct filestat_list* files    = procs    ? procstat_getfiles(procstat, procs, true)                 : NULL;
+        if ( files ) {
+            filestat* entry;
+            STAILQ_FOREACH(entry, files, next) {
+                std::string path(entry->fs_path);
+                pushPath(path,libs,paths);
+            }
+        }
+        // free resources
+        if ( files    ) procstat_freefiles(procstat, files);
+        if ( procs    ) procstat_freeprocs(procstat, procs);
+        if ( procstat ) procstat_close    (procstat);
+
+    #elif defined(__unix__)
+        // read file /proc/self/maps which has a list of files in memory
+        std::ifstream maps("/proc/self/maps",std::ifstream::in);
```

Comment:
You will need to include the header `<fstream>`, right now the linux jobs are failing with this error:

```
/home/travis/build/Exiv2/exiv2/src/futils.cpp: In function ‘std::vector<std::__cxx11::basic_string<char> > Exiv2::getLoadedLibraries()’:
/home/travis/build/Exiv2/exiv2/src/futils.cpp:551:27: error: variable ‘std::ifstream maps’ has initializer but incomplete type
         std::ifstream maps("/proc/self/maps",std::ifstream::in);
```